interview-questions.md

```diff
@@ -7,43 +7,33 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is an instance variable in Ruby? How is it different from other variables?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: An instance variable is a variable that starts with @ symbol and belongs to object, wheres class variables belong only to the class.
 
+  Researched answer: An instance variable starts with a @ symbol as well initialized to the value of nil. Instance variables belong to the object itself. They can be accessed throughout the class as well.
 
 2. What is a block in Ruby?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: Blocks in Ruby are anonymous functions that allow you to 'do' something either a certain amount of times or iterating through an array as well as touch or do something with every element in an array
 
+  Researched answer: Blocks in Ruby are anonymous functionsthat are created using the keywords do/end or with {}. Blocks do not require a name and they act like a method in the sense that you can pass in arguemtns to get a value back. Blocks are always passed to a method call.
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: An implicit returns means that the return is implied which means that the user doesn't have to say return when returning a variable.
 
+  Researched answer: Every block will return the value at the last line automatically. You can still explicitly use the return statement but it might cause some errors for the user later on. 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: OOP is a programming concept that revoloes around the idea of having objects. Whereas functional programming is a concept that revlolves around constructing programs from creating functions.
 
+  Researched answer: Funtional programming is a declarative programming in which functions are trees of expressions. It has come from lambda calculus where computations are only functions. OOB is a programming language that has objects that contain data where they can interact with one another. The most popular ones are class based which means that the objects are instances of the class created.
```

Comment:
Awesome! if you find this topic interesting there are other kinds of languages too -- like prototypical languages (ex. javascript)